### ISSUE TITLE
Mismatched valid ranges of parameters between documentation and source code 

### ISSUE THREAD

**Issue by KimHyungSub** [PX4/PX4-Autopilot#17909]
I found three mismatched valid ranges of parameters between documentation and source code, when PX4 enforces reasonable constraints.

1) IMU_GYRO_RATEMAX
- Valid range based on documentation: 100 - 2000 (https://docs.px4.io/master/en/advanced_config/parameter_reference.html)
- Valid range based on source code: 100 - 4000 (https://github.com/PX4/PX4-Autopilot/blob/a728431e4350917f3add9814732715101ac99b4d/src/lib/drivers/accelerometer/PX4Accelerometer.hpp#L51)

2) IMU_GYRO_RATEMAX 
- Valid range based on documentation: 100 - 2000 (https://docs.px4.io/master/en/advanced_config/parameter_reference.html)
- Valid range based on source code: 400 - 5000 (https://github.com/PX4/PX4-Autopilot/blob/f9d8c613b048f58eb3110e9af13cb3a89c4c866f/src/modules/sih/sih.cpp#L90)

3) _param_imu_gyro_ratemax is not limited --> imu_integration_rate_hz is also not limited --> fail to enforce reasonable constraint
- https://github.com/PX4/PX4-Autopilot/blob/d87cd171f4461ee0525052d9e0f15cc3897ad7ae/src/modules/sensors/vehicle_imu/VehicleIMU.cpp#L122

Thanks!

**Comment by dagar** [comment]
Good catch, although some of this was actually intentional (and may need to be rethought or better documented).

1) This is the "real" hard limit that I thought would be more than fast enough for nearly any case, but also aligned well with what a large portion of the common IMUs are doing (8 kHz gyro + 4 kHz accel).
2) This is more of a soft limit that can be overridden if needed. The problem is the actual limit depends on the specific board and other things you might be using. Each high rate IMU is pretty expensive, and boards have between 1-3 IMUs. Older boards might be fine with 1 or 2 IMUs at 1-2 kHz, newer STM32H7s can handle all 3 IMUs at 2-4 kHz. If you set the rate too high it might still work, but will starve lower priority things in the system (mavlink, etc).
3) This IMU integration rate is what feeds the estimator output predictor (attitude estimate publication rate). I didn't see the need for this to be faster than 1000 Hz.

Does that make sense?

If we added a mechanism to more gracefully handle the system being overloaded we could approach this a bit differently.
